src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -143,6 +143,8 @@ inline void sockinfo_tcp::init_pbuf_custom(mem_buf_desc_t *p_desc)
 	p_desc->lwip_pbuf.pbuf.type = PBUF_REF;
 	p_desc->lwip_pbuf.pbuf.next = NULL;
 	p_desc->lwip_pbuf.pbuf.payload = (u8_t *)p_desc->p_buffer + p_desc->rx.tcp.n_transport_header_len;
+	/* Override default free function to return rx pbuf to the CQ cache */
+	p_desc->lwip_pbuf.custom_free_function = sockinfo_tcp::tcp_rx_pbuf_free;
```

Comment:
I prefer to explicitly change `pbuf_free()` with 'tcp_rx_pbuf_free()' and use the same change as was done for `tcp_rx_pbuf_free()`. It is more save for future.
The reason: custom_free_function  callback is common for tx and rx flow and it is easy to do mistake in future change